UserPreferences/UserPreferences.cpp

```diff
@@ -66,59 +74,443 @@ namespace WPEFramework {
+                    LOGINFO("Successfully obtained UserSettings interface after %d retries", count);
+                    break;
+                } else {
+                    count++;
+                    LOGERR("Failed to get UserSettings interface, retry: %d/%d", count, RETRY_COUNT);
+                    usleep(RETRY_INTERVAL_MS * 1000);
+                }
+            } while (count < RETRY_COUNT);
+           
+
+            if (nullptr == userSettings) {
+                LOGERR("Failed to obtain UserSettings interface after %d retries", RETRY_COUNT);
+                return " ";
+            }
+
+            PerformMigration(*userSettings);
+        
+            userSettings->Register(&_notification);
+            LOGINFO("Successfully registered for UserSettings notifications");
+
+            userSettings->Release();
+
+            return {};
+        }
+
+        void UserPreferences::Deinitialize(PluginHost::IShell* /* service */) {
             LOGINFO("Deinitialize");
+            Exchange::IUserSettings* userSettings = _service->QueryInterfaceByCallsign<Exchange::IUserSettings>("org.rdk.UserSettings");
+            if (nullptr != userSettings) {
+                userSettings->Unregister(&_notification);
+                userSettings->Release();
+                userSettings = nullptr;
+            }
+            _adminLock.Lock();
+            if (nullptr != _service) {
+                _service->Release();
+                _service = nullptr;
+            }
+            _adminLock.Unlock();
+
             UserPreferences::_instance = nullptr;
         }
 
+        string UserPreferences::Information() const {
+            return "This UserPreferences Plugin stores and retrieves settings using the UserSettings Plugin";
+        }
+
+        void UserPreferences::Notification::OnPresentationLanguageChanged(const string& language) {
+            /* 
+            * Executing in the UserSettings notification context because:
+            * 1. This callback is triggered only when the presentation language is changed, 
+            *    which is a relatively infrequent operation and not performance critical.
+            * 2. The handler does not make any blocking or recursive calls back into UserSettings, 
+            *    so there is no risk of deadlock or circular dependency.
+            *
+            * Therefore, there's no need to offload this work to another thread. 
+            * Handling it directly in the caller context ensures simplicity, avoids unnecessary thread management, 
+            * and is safe within the constraints of this use case.
+            */
+             _parent->OnPresentationLanguageChanged(language);
+        }
+
+        void UserPreferences::OnPresentationLanguageChanged(const string& language) {
+            LOGINFO("Presentation language changed to: %s", language.c_str());
+            string uiLanguage;
+            if (ConvertToUserPrefsFormat(language, uiLanguage)) {
```

Comment:
can we optimize and update only if language has changed ( something like if (language != _lastUILanguage)